### ISSUE TITLE
Token issue potential bug

### ISSUE THREAD

**Issue by mattheworiordan** [ably/ably-java#76]
We have identified a situation where a client library could, in theory, end in an endless loop of trying to reissue tokens and reattempt requests.  We've addressed this now in the spec at https://github.com/ably/docs/commit/b628023b83aca480464d55e4260e21ee20690d7a.

This library needs to be updated accordingly.

**Comment by paddybyers** [comment]
This is addressed for HTTP as a result of http auth changes in https://github.com/ably/ably-java/pull/123.

Need to check ws still.

**Comment by mattheworiordan** [comment]
@paddybyers @trenouf can we close this, or can this still affect WS?

**Comment by mattheworiordan** [comment]
@paddybyers @psolstice does this issue still apply?

**Comment by AndyNicks** [comment]
@paddybyers does this issue still apply or we can close it?

**Comment by paddybyers** [comment]
I don't think we confirmed - we should keep this issue open and investigate when someone is able to pick it up
